lib/snappy_ex.ex

```diff
@@ -0,0 +1,63 @@
+defmodule LambdaEthereumConsensus.SnappyEx do
+  @moduledoc """
+    SSZ library in Elixir
+  """
+
+  def decompress_frames(chunks) do
+    with {:ok, chunks} <- validate_stream(chunks) do
+      decompress_frames(chunks, <<>>)
+    end
+  end
+
+  defp decompress_frames(chunks, acc) do
+    if byte_size(chunks) != 0 do
+      {chunk_id, chunks} = process_chunk_id(chunks)
+      {chunks_size, chunks} = process_chunk_size(chunks)
+      {acc, chunks} = process_chunks(chunks, acc, chunks_size, chunk_id)
+      decompress_frames(chunks, acc)
```

Comment:
We should make all these return `{:ok, result}` and match on them using `with`, to handle errors.